PoC-updater/updater.go

```diff
@@ -73,26 +91,65 @@ func process(storage Storage) {
 	c := types.ClusterEntry{
 		OrgID:       testedOrgID,
 		ClusterName: testedCluster,
+		UpdatedAt:   timestamp,
 	}
 
-	// read older reports for a cluster
-	reported, err := storage.ReadLastNNotificationRecords(c, 10)
+	// try to read older report for a cluster
+	reported, err := storage.ReadLastNNotificationRecords(c, 1)
 	if err != nil {
 		log.Error().Err(err).Msg("Read last n reports")
 	}
-	for _, r := range reported {
-		rOldAsString := r.Report
-		log.Info().Time("notified at", time.Time(r.NotifiedAt)).Msg("notified record")
 
-		notify, err := CompareReports(rOldAsString, rNewAsString)
-		if err != nil {
-			log.Error().Err(err).Msg("Unable to compare")
-		}
-		log.Info().Bool("resolution", notify).Msg("Should notify user")
-		fmt.Println()
+	notifiedAt := types.Timestamp(time.Now())
+
+	// check if the new result has been stored for brand new cluster
+	if len(reported) == 0 {
+		log.Info().Str("cluster", string(c.ClusterName)).Msg("New cluster -> send instant report")
+		processNotificationForCluster(storage, c, rNewAsString, notificationTypes, states, notifiedAt)
+		// and we are done!
+		return
+	}
+
+	// it is not a brand new cluster -> compare new report with older one
+	rOldAsString := reported[0].Report
+	notify, err := CompareReports(rOldAsString, rNewAsString)
+	if err != nil {
+		log.Error().Err(err).Msg("Unable to compare old and new reports")
+	}
+	log.Info().Bool("resolution", notify).Msg("Should notify user")
+
+	// if new report differs from the older one -> send notification
+	if notify {
+		log.Info().Str("cluster", string(c.ClusterName)).Msg("Different report from the last one")
+		processNotificationForCluster(storage, c, rNewAsString, notificationTypes, states, notifiedAt)
+		// and we are done!
+	} else {
+		log.Info().Str("cluster", string(c.ClusterName)).Msg("Same report as before")
+		// store notification info about not sending the notification
+		storage.WriteNotificationRecordForCluster(c, notificationTypes.Daily, states.SameState, rNewAsString, notifiedAt, "")
+	}
+}
+
+func processNotificationForCluster(storage Storage, c types.ClusterEntry,
+	rNewAsString types.ClusterReport,
+	notificationTypes types.NotificationTypes, states types.States,
+	notifiedAt types.Timestamp) {
+	instantIssue, err := fakeProcessReport(rNewAsString)
+	if err != nil {
+		storage.WriteNotificationRecordForCluster(c, notificationTypes.Daily, states.ErrorState, rNewAsString, notifiedAt, err.Error())
+	}
+
+	if instantIssue {
+		storage.WriteNotificationRecordForCluster(c, notificationTypes.Daily, states.SentState, rNewAsString, notifiedAt, "")
+	} else {
+		storage.WriteNotificationRecordForCluster(c, notificationTypes.Daily, states.SameState, rNewAsString, notifiedAt, "")
```

Comment:
notificationTypes.Weekly

Reply:
nope, it means that the new report is the same as already reported one, so we skip it for now

Reply:
Ah! Definitely makes sense.
So maybe renaming the `instantIssue` variable will clarify this if-else block.